TI-reports/2024/2024-Q3-BEST-WG.md

```diff
@@ -0,0 +1,130 @@
+- Security Baseline - [link](
+  
+### Leads	
+- WG - CRob  
+- BP Badge and SecDev course - David Wheeler
+- Compiler Hardening Guides - Thomas Nyman & Geog Kunz
+- EDU SIG - CRob & Dave Russo
+- Mem Safety SIG - Nell Shamrell-Harrignton & Avishay Balter
+- Python Hardening Guide - Helge & Georg
+- Scorecard - Laurent Simon & Stephen Augustus
+- Security Baseline - Eddie Knight
+- WebDev Sec BP - Daniel Appelquist
+
+## Activity
+### Best Practices Badge
+#### Purpose
+- The Open Source Security Foundation (OpenSSF) Best Practices badge is a way for Free/Libre and Open Source Software (FLOSS) projects to show that they follow best practices. Projects can voluntarily self-certify, at no cost, by using this web application to explain how they follow each best practice.
+#### Current Status
+-
+- #### Up Next
+- TBD
+
+### Concise Guides
+#### Purpose
+- Artifacts that consolidate BEST practices in OSS software development and management techniques
+#### Current Status
+- Continued revisions, updates, & enhancements to these core guides
+#### Up Next
+- TBD
+
+### EDU.SIG
+#### Purpose
+- Deliver Baseline Secure Software Development Education and Certification to All.  Provide access to open and widely available education materials to all learners.
+Materials will be maximally accessible and easy to consume for all learners.
+#### Current Status
+- Many simultaneous activities
+- Recent release of LF Research study on Security Edutation for Developers
+- Academic Accredidation team working on kicking off program to "certify" collegiate programs that meet OpenSSF & CNCF best practices
+- Security for Developer Managers class progressing into two pieces of collateral: Manager class & terms-definitions
+#### Up Next
+- Security Architect class outline reviewed and content development will come next
+- "201 level" class will come after
+- 
+### Memory Safety SIG
+#### Purpose
+- The Memory Safety SIG is a group working within the OpenSSF's Best Practices Working Group formed to advance and deliver upon The OpenSSF's Mobilization Plan - Stream 4.
+#### Current Status
+-Have drafted a “Memory Safety Continuum” concept document
+- Have gathered guides/practices related to best memory safety practices in both memory safe by default and non memory safe by default languages
+#### Up Next 
+- Produce a Memory Safety workshop (modeled after W3C workshops). Theme is “Improving Memory Safety in an Imperfect World”
+- Finalize Memory Safety Continuum doc
+
+### Python Hardening Guide
+#### Purpose
+
+#### Current Status
+
+#### Up Next
+
+### Scorecard
+#### Purpose
+-To help open source maintainers improve their security best practices and to help open source consumers judge whether their dependencies are safe.
+- Scorecard is an automated tool that assesses a number of important heuristics ("checks") associated with software security and assigns each check a score of 0-10.
+#### Current Status
+
+#### Up Next
```

Comment:
Inside the OpenSSF, I feel like people generally understand that Scorecard is a framework for **maintainers** to walk them through improving their secure software development practices - sort of a gamified maturity model, if you will.

Outside the OpenSSF, we've seen a fair amount of confusion as to if the Scorecard scores should be used by **consumers** of open source software as a way to boil down risk assessment into a single number, the most recent example of this being https://openssf.slack.com/archives/C019M98JSHK/p1720098402786119.

I think the OpenSSF is aligned on that Scorecard is meant for maintainers and not consumers. Have we thought about how we might adjust our messaging to prevent this confusion in the future? I'm specifically thinking of places like https://github.com/ossf/scorecard and https://securityscorecards.dev/, but I'm open to other suggestions!

Reply:
Counter point: I've seen enterprises seek to understand scorecard/scores as a way to automate risk when they consume OSS, and create mechanisms to automate OSS ingestion. The Metrics API SIG (under the former Metrics and Metadata WG) was creating an API where enterprises could pull Scorecard and other data into their processes.

Reply:
Relevant scorecard issue: https://github.com/ossf/scorecard/issues/4219

Reply:
Err... The README *starts* with:

> We created Scorecard to help open source maintainers improve their security best practices and to help open source consumers judge whether their dependencies are safe.